java-agent/instrumentation/servlet/request-2/src/main/java/io/opentelemetry/auto/instrumentation/servlet2/Servlet2Advice.java

```diff
@@ -24,13 +23,14 @@
 public class Servlet2Advice {
 
   @Advice.OnMethodEnter(suppress = Throwable.class)
-  public static AgentScope onEnter(
+  public static SpanScopePair onEnter(
       @Advice.This final Object servlet,
       @Advice.Argument(0) final ServletRequest request,
       @Advice.Argument(value = 1, readOnly = false, typing = Assigner.Typing.DYNAMIC)
           ServletResponse response) {
-    final boolean hasActiveTrace = activeSpan() != null;
-    final boolean hasServletTrace = request.getAttribute(SPAN_ATTRIBUTE) instanceof AgentSpan;
+    final Span current = TRACER.getCurrentSpan();
+    final boolean hasActiveTrace = current != null && current.getContext().isValid();
```

Comment:
```suggestion
    final boolean hasActiveTrace = current.getContext().isValid();
```
TRACER.getCurrentSpan() never returns `null`